internal/tools/pod_exec_cmd.go

```diff
@@ -0,0 +1,159 @@
+package tools
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/strowk/mcp-k8s-go/internal/k8s"
+	"github.com/strowk/mcp-k8s-go/internal/utils"
+
+	"github.com/strowk/foxy-contexts/pkg/fxctx"
+	"github.com/strowk/foxy-contexts/pkg/mcp"
+	"github.com/strowk/foxy-contexts/pkg/toolinput"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+func NewPodExecCommandTool(pool k8s.ClientPool) fxctx.Tool {
+	k8sNamespace := "namespace"
+	k8sPodName := "podName"
+	execCommand := "command"
+	schema := toolinput.NewToolInputSchema(
+		toolinput.WithString(k8sNamespace, "The name of the namespace where the pod to execute the command is located."),
+		toolinput.WithString(k8sPodName, "The name of the pod in which the command needs to be executed."),
+		toolinput.WithString(execCommand, "The command to be executed inside the pod."),
+	)
+	return fxctx.NewTool(
+		&mcp.Tool{
+			Name:        "pod-exec-command",
+			Description: utils.Ptr("Enter the specified namespace's pod, execute the specified command, and return the execution result."),
+			InputSchema: schema.GetMcpToolInputSchema(),
+		},
+		func(args map[string]interface{}) *mcp.CallToolResult {
+			input, err := schema.Validate(args)
+			if err != nil {
+				return errResponse(err)
+			}
+			k8sNamespace := input.StringOr(k8sNamespace, "")
+			k8sPodName := input.StringOr(k8sPodName, "")
+			execCommand := input.StringOr(execCommand, "")
+
+			config := getConfigFromKubeFile(os.Getenv("KUBECONFIG"))
+			output, err := cmdExecuter(config, k8sPodName, k8sNamespace, execCommand)
+			if err != nil {
+				return errResponse(err)
+			}
+
+			var content mcp.TextContent
+			contents := []interface{}{}
+			content, err = NewJsonContent(ExecResult{
+				ExecResultKey:   "podName",
+				ExecResultValue: output[0],
+			})
+			if err != nil {
+				return errResponse(err)
+			}
+			contents = append(contents, content)
+			content, err = NewJsonContent(ExecResult{
+				ExecResultKey:   "command",
+				ExecResultValue: output[1],
+			})
+			if err != nil {
+				return errResponse(err)
+			}
+			contents = append(contents, content)
+			content, err = NewJsonContent(ExecResult{
+				ExecResultKey:   "stdout",
+				ExecResultValue: output[2],
+			})
+			if err != nil {
+				return errResponse(err)
+			}
+			contents = append(contents, content)
+			content, err = NewJsonContent(ExecResult{
+				ExecResultKey:   "stdin",
+				ExecResultValue: output[3],
+			})
+			if err != nil {
+				return errResponse(err)
+			}
+			contents = append(contents, content)
```

Comment:
I would suggest to simply have one structure  with two strings - "stderr" and "stdout" , f.e:

```golang
type ExecResult struct {
	Stdin string          `json:"stdin"`
	Stderr string        `json:"stderr"`
}
```
And give such structure to NewJsonContent.

This would remove from context window four strings "execResultKey", "execResultValue", .. etc, but would still keep results understandable.

An alternative that could improve readability of result could be simply return four TextContent's - firstly line "stdout:", then everything that is in stdout, then same for stderr. Due to absence of escapings, this might be actually more compact for certain inputs, while possibly result in better structure in some clients (since no extra wrapping).